src/videojs-contrib-hls.js

```diff
@@ -52,13 +52,20 @@ const Hls = {
   xhr
 };
 
+let GOAL_BUFFER_LENGTH = 30;
+
 Object.defineProperty(Hls, 'GOAL_BUFFER_LENGTH', {
   get() {
-    videojs.log.warn('Hls.GOAL_BUFFER_LENGTH should not be used');
-    return 30;
+    videojs.log.warn('Hls.GOAL_BUFFER_LENGTH is deprecated and should not be used');
+    return GOAL_BUFFER_LENGTH;
   },
   set(v) {
-    videojs.log.warn('Hls.GOAL_BUFFER_LENGTH should not be used and will not be set');
+    videojs.log.warn('Hls.GOAL_BUFFER_LENGTH is deprecated and should not be used');
+    if (typeof v !== 'number') {
```

Comment:
I know that later on we use options.GOAL_BUFFER_LENGTH || 30. In the case of 0, we will accept the value, but ignore it later. I know that no one should be using 0, and it might be somewhat silly to guard against, but we can ensure v > 0 and warn below that it must be a number greater than 0.